components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+
+// Common components
+import { SettingsText } from '../../../components/default'
+import { Toggle } from '../../../components/toggle'
+
+// Assets
+import InfoIcon from '../../../components/default/settings/assets/info-icon.svg'
+
+// Utilities
+import { getLocale } from '../../../../common/locale'
+
+const isDarkTheme = (p: any) => {
+  return p.theme.name === 'Brave Dark'
+}
+
+interface Props {
+  onChange: () => void
+  onEnableRewards: () => void
+
+  checked: boolean
+  disabled: boolean
+  rewardEnabled: boolean
+  adsEnabled: boolean
+  canSupportAds: boolean
+}
+
+const Container = styled('div')<{}>`
+  background-color: rgba(93, 181, 252, 0.2);
+  border-radius: 8px;
+  padding: 8px;
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
+`
+
+const ToggleRow = styled('div')<{}>`
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: 12px 14px;
+  border-radius: 10px;
+  background-color: ${p => isDarkTheme(p) ? css`#1E2029` : css`white`};
```

Comment:
```suggestion
  background-color: var(--background1);
```

should automatically handle light/dark mode (and you can remove `isDarkTheme`)

Reply:
See `brave/components/web-components/app.global.scss` for the CSS variables we have. I think there's a convention for transforming the color name in Figma to the css variable name, but there are a bunch of missing colors in there sadly.

Reply:
> convention for transforming the color name in Figma to the css variable name

Great! Is there a doc for this part? I was wondering how we manage all these variable stuffs. Also this is my first time to use Figma, It'd be really nice to learn best practices.

Reply:
I haven't really used it much either :laughing: I think in the new design system the Figma ==> CSS variable transformation is more straightforward. I often just search for the hex color in `app.global.css` or the last bit of the color name in figma but without leading `0s` (ie. `Light Theme/Text/text01` becomes `--text1`).

Reply:
> Theme/Text/text01 becomes --text1).

Aha! that's the convention! thanks a lot :)